### ISSUE TITLE
[Bug]: Picked up items dont stack correctly.

### ISSUE THREAD

**Issue by pafeuu** [someaddons/recipeessentials#13]
### Describe the bug you're experiencing

![image](https://github.com/user-attachments/assets/f6340889-3ca1-491f-8e01-561b7c40b312)
When picking up blocks they are often pushed into new slot instead of stacking with each other.

### Reproducability

Try chopping a tree or mine a bit of stone

### Mod up to date

- [X] Before submitting this issue I updated to the newest version and reproduced it

### Minecraft version

1.21

### Modloader version

Fabric

### Logs

https://gist.github.com/pafeuu/1deac9513ad6e0c5f2cdb50ef00019ed

**Comment by someaddons** [comment]
cannot reproduce it with just recipe essentials, so probably a mod conflict of some kind. Which modpack are you using?

**Comment by pafeuu** [comment]
[FabricTest.zip](https://github.com/user-attachments/files/17030736/FabricTest.zip)
I tried making something of my own here is a modlist. Fair warning that included some mods available only on modrinth and I didnt know how else to send this. Here is a list of mods from modrinth:
-Charm
-Axiom
-Additional entity attributes
-Flashback
-HorseBuff
-Strange
I wanted to rule those out of mod conflicts so I removed them and now I wasn't even able to create a world. Removing recipe essentials fixed that issue
[crash-2024-09-17_17.26.06-client.txt](https://github.com/user-attachments/files/17030795/crash-2024-09-17_17.26.06-client.txt)
Here is the crash log from situation above.
